account_manager/src/validator/exit.rs

```diff
@@ -0,0 +1,440 @@
+        &client,
+        &spec,
+        stdin_inputs,
+        &testnet_config,
+    ))?;
+
+    Ok(())
+}
+
+/// Gets the keypair and validator_index for every validator and calls `publish_voluntary_exit` on it.
+async fn publish_voluntary_exits<E: EthSpec>(
+    validator_dirs: Vec<ValidatorDir>,
+    secrets_dir: &PathBuf,
+    client: &BeaconNodeHttpClient,
+    spec: &ChainSpec,
+    stdin_inputs: bool,
+    testnet_config: &Eth2TestnetConfig<E>,
+) -> Result<(), String> {
+    let genesis_data = get_geneisis_data(client).await?;
+    let testnet_genesis_root = testnet_config
+        .genesis_state
+        .as_ref()
+        .expect("network should have valid genesis state")
+        .genesis_validators_root;
+
+    // Verify that the beacon node and validator being exited are on the same network.
+    if genesis_data.genesis_validators_root != testnet_genesis_root {
+        return Err(
+            "Invalid genesis state. Please ensure that your beacon node is on the same network \
+                 as the validator you are publishing an exit for"
+                .to_string(),
+        );
+    }
+    let epoch = get_current_epoch::<E>(genesis_data.genesis_time, spec)
+        .map_err(|e| format!("Failed to get current epoch: {:?}", e))?;
+
+    let fork = get_beacon_state_fork::<E>(client, epoch).await?;
+
+    for validator_dir in validator_dirs {
+        let keypair = load_voting_keypair(&validator_dir, secrets_dir, stdin_inputs)?;
+        let validator_index = get_validator_index::<E>(client, &keypair.pk, epoch, spec).await;
+
+        match validator_index {
+            Ok(index) => {
+                if let Err(e) = publish_voluntary_exit::<E>(
+                    &keypair,
+                    client,
+                    spec,
+                    stdin_inputs,
+                    epoch,
+                    index,
+                    &fork,
+                    genesis_data.genesis_validators_root,
+                )
+                .await
+                {
+                    eprintln!(
+                        "Failed to publish voluntary exit for validator {:?}: {}",
+                        validator_dir.dir(),
+                        e
+                    );
+                }
+            }
+            Err(e) => eprintln!("{}", e),
+        }
+    }
+    Ok(())
+}
+
+/// Constructs a `VoluntaryExit` object for a given validator and signs it using the given bls keypair.
+/// Publishes the voluntary exit to the beacon chain using the beacon node endpoint.
+#[allow(clippy::too_many_arguments)]
+async fn publish_voluntary_exit<E: EthSpec>(
+    keypair: &Keypair,
+    client: &BeaconNodeHttpClient,
+    spec: &ChainSpec,
+    stdin_inputs: bool,
+    epoch: Epoch,
+    validator_index: u64,
+    fork: &Fork,
+    genesis_validators_root: Hash256,
+) -> Result<(), String> {
+    let voluntary_exit = VoluntaryExit {
+        epoch,
+        validator_index,
+    };
+
+    let signed_voluntary_exit =
```

Comment:
Just to be super cautious, I would suggest creating this *after* the confirmation.